lwcomponent/catalog.go

```diff
@@ -279,63 +278,69 @@ func NewCatalog(
 			}
 		}
 
-		api := NewAPIInfo(c.Id, c.Name, ver, allVersions, c.Description, c.Size, c.Deprecated, Type(c.ComponentType))
+		apiInfo := NewAPIInfo(c.Id, c.Name, ver, allVersions, c.Description, c.Size, c.Deprecated, Type(c.ComponentType))
+		cdkComponents[c.Name] = NewCDKComponent(c.Name, c.Description, Type(c.ComponentType), apiInfo, nil)
+	}
 
-		host, found := localComponents[c.Name]
-		if found {
-			delete(localComponents, c.Name)
-		}
+	components, err := LoadComponents(cdkComponents)
+	if err != nil {
+		return nil, err
+	}
 
-		component := NewCDKComponent(c.Name, c.Description, Type(c.ComponentType), api, host)
+	return &Catalog{client, components, stageConstructor}, nil
+}
 
-		cdkComponents[c.Name] = component
+func NewCachedCatalog(
+	client *api.Client,
+	stageConstructor StageConstructor,
+	cachedComponentsApiInfo map[string]ApiInfo,
+) (*Catalog, error) {
+	if stageConstructor == nil {
+		return nil, errors.New("nil Catalog StageConstructor")
 	}
 
-	for _, localHost := range localComponents {
-		if localHost.Development() {
-			devInfo, err := NewDevInfo(localHost.Dir())
-			if err != nil {
-				return nil, err
-			}
+	cachedComponents := make(map[string]CDKComponent, len(cachedComponentsApiInfo))
 
-			cdkComponents[localHost.Name()] = NewCDKComponent(
-				localHost.Name(),
-				devInfo.Desc,
-				devInfo.ComponentType,
-				nil,
-				localHost)
-		} else {
-			// @jon-stewart: persisted API info
-			cdkComponents[localHost.Name()] = NewCDKComponent(localHost.Name(), "", BinaryType, nil, localHost)
-		}
+	for _, a := range cachedComponentsApiInfo {
+		a := a
+		cachedComponents[a.Name] = NewCDKComponent(a.Name, a.Desc, a.ComponentType, &a, nil)
 	}
 
-	return &Catalog{client, cdkComponents, stageConstructor}, nil
-}
+	components, err := LoadComponents(cachedComponents)
+	if err != nil {
+		return nil, err
+	}
 
-func LocalComponents() (components []CDKComponent, err error) {
-	var localHost map[string]HostInfo
+	return &Catalog{client, components, stageConstructor}, nil
+}
 
-	localHost, err = loadLocalComponents()
+// LoadComponents combines the passed in components with the local components
+func LoadComponents(components map[string]CDKComponent) (allComponents map[string]CDKComponent, err error) {
```

Comment:
Why is this function public? Also, the function name doesn't reflect what the function is actually doing, I would recommend renaming it to 
```suggestion
func mergeComponents(components map[string]CDKComponent) (allComponents map[string]CDKComponent, err error) {
```
or something like that

Reply:
Updated.